src/parser/helper/Helper.test.ts

```diff
@@ -0,0 +1,140 @@
+        });
+
+        it("should create UNIX-style relative paths when configured", () => {
+            const filePath = "/some/path/for/the/test.extension";
+            const config = new Configuration(
+                "/some/path",
+                "invalid/output/path",
+                false,
+                "",
+                false,
+                "",
+                false,
+                true,
+                false,
+            );
+
+            expect(formatPrintPath(filePath, config, path.posix)).toEqual("for/the/test.extension");
+        });
+
+        it("should create DOS-style relative paths when configured", () => {
+            const filePath = "C:\\Users\\user\\documents\\code\\more-code\\file.extension";
+            const config = new Configuration(
+                "C:\\Users\\user\\documents\\code",
+                "invalid/output/path",
+                false,
+                "",
+                false,
+                "",
+                false,
+                true,
+                false,
+            );
+
+            expect(formatPrintPath(filePath, config, path.win32)).toEqual(
+                "more-code\\file.extension",
+            );
+        });
+
+        it("should return the file name if the file path equals the sources path (UNIX-style)", () => {
+            const filePath = "/some/path/for/the/test.extension";
+            const config = new Configuration(
+                "/some/path/for/the/test.extension",
+                "invalid/output/path",
+                false,
+                "",
+                false,
+                "",
+                false,
+                true,
+                false,
+            );
+
+            expect(formatPrintPath(filePath, config, path.posix)).toEqual("test.extension");
+        });
+
+        it("should return the file name if the file path equals the sources path (DOS-style)", () => {
+            const filePath = "C:\\Users\\user\\documents\\code\\more-code\\file.extension";
+            const config = new Configuration(
+                "C:\\Users\\user\\documents\\code\\more-code\\file.extension",
+                "invalid/output/path",
+                false,
+                "",
+                false,
+                "",
+                false,
+                true,
+                false,
+            );
+
+            expect(formatPrintPath(filePath, config, path.win32)).toEqual("file.extension");
+        });
+
+        it("should both create relative paths and replace forward slashes when configured", () => {
+            const filePath = "/some/path/for/the/test.extension";
+            const config = new Configuration(
+                "/some/path",
+                "invalid/output/path",
+                false,
+                "",
+                false,
+                "",
+                false,
+                true,
+                true,
+            );
```

Comment:
It is hard to see for what which argument is.
Maybe this could help::
const config = `{ sourcePath: "a/b/c", otherAtrribute: false } as Configuration`. 

I am not sure though if this compiles :-)

Reply:
I will look into it.

Reply:
If it works, it would bypass the constructor of Configuration which does actually do more things than just assinging values. That would be really unelegant.

I have looked around a bit, and using an interface instead for the parameters is a choice if one wants to have named parameters.

Reply:
I have now introduced the interface `ConfigurationParams` for this.